contracts/clients/src/BlsSigner.ts

```diff
@@ -464,6 +390,54 @@ export default class BlsSigner extends Signer {
     throw new Error("_legacySignMessage() is not implemented");
   }
 
+  async _validateTransaction(
+    transaction: Deferrable<ethers.providers.TransactionRequest>,
+  ): Promise<ethers.providers.TransactionRequest> {
+    const resolvedTransaction = await ethers.utils.resolveProperties(
+      transaction,
+    );
+
+    if (!resolvedTransaction.to) {
+      throw new TypeError("Transaction.to should be defined");
+    }
+
+    if (!resolvedTransaction.from) {
+      resolvedTransaction.from = await this.getAddress();
```

Comment:
This is a subtle, but important change, it ensures we don't have to require the following logic in many places:

```ts
if (!resolvedTransaction.from) {
   throw new TypeError("Transaction.from should be defined");
}
```

This is because the provider `estimateGas` method now requires `from` to calculate the nonce, and the transaction that this method is validating will get pass to that method later in the send transaction methods. See changes to estimateGas for more context. If that doesn't make sense can explain in more detail